### PR TITLE
Added UI config service

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**

### CODE REVIEW

File: src/app/public/modules/ui-config/ui-config.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs';
+
+@Injectable()
+export class SkyUIConfigService {
+
+  public getConfig(key: string, defaultConfig?: any): Observable<any> {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/ui-config/ui-config.service.ts]: Curious, does `defaultConfig` need a type?

File: src/app/public/modules/ui-config/ui-config.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs';
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/ui-config/ui-config.service.ts]: Is it “safe” to import from rxjs blindly? I seem to remember an article about doing this in  rxjs version 4 but could be wrong…
```
import {
  Observable
} from 'rxjs'; // Should be `rxjs/Observable`?
```

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/ui-config/ui-config.service.ts]: Related: https://github.com/angular/angular/issues/20349

File: src/app/public/modules/ui-config/ui-config.service.ts (hunk 1 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/ui-config/ui-config.service.ts]: `defaultConfig` can be anything the consumer wants to store as a setting, so it needs to be of type `any`.

File: src/app/public/modules/ui-config/ui-config.service.ts (hunk 2 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/ui-config/ui-config.service.ts]: Fixed.

**Pull request merged**